src/lib/PriorityOrderLib.sol

```diff
@@ -0,0 +1,106 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity ^0.8.0;
+
+import {OrderInfo, InputToken, OutputToken} from "../base/ReactorStructs.sol";
+import {OrderInfoLib} from "./OrderInfoLib.sol";
+import {ERC20} from "solmate/src/tokens/ERC20.sol";
+
+struct PriorityInput {
+    ERC20 token;
+    uint256 amount;
+    // the less amount of input to be received per wei of priority fee
+    uint256 mpsPerPriorityFeeWei;
+}
+
+struct PriorityOutput {
+    address token;
+    uint256 amount;
+    // the extra amount of output to be paid per wei of priority fee
+    uint256 mpsPerPriorityFeeWei;
+    address recipient;
+}
+
+/// @dev External struct used to specify priority orders
+struct PriorityOrder {
+    // generic order information
+    OrderInfo info;
+    // the block at which the order becomes active
+    uint256 startBlock;
```

Comment:
I think this is good for now but one thing to consider for later --

this is signed over by the user, and users take unknown amt of sign to sign. In order to make sure this is in the future by the time the order is signed we will likely need to buffer quite a bit. May be difficult to tightly optimize for execution latency without some kind of cosigner situation just for updating the block

Reply:
potentially in the future users specify `minBlock` and `maxBlock` and labs can optionally cosign `targetBlock` with higher precision